players/base.py

```diff
@@ -357,7 +357,11 @@ def has_overdrive(self, combat=False) -> bool:
     def is_turn(self) -> bool:
         return memory.main.get_battle_char_turn() == self.id
 
-    def in_danger(self, danger_threshold, combat=False) -> bool:
+    def in_danger(self, danger_threshold) -> bool:
+        if memory.main.battle_active():
```

Comment:
This check is now happening twice. It would be worth adding a method or property on the `Player` class to tell if we are in combat or not. Something like:

```python
@property
def in_combat(self):
    return memory.main.battle_active()

...

def in_danger(self, danger_threshold) -> bool:
    combat = self.in_combat
    ...
```

Reply:
Updated, please check again.